Wrappers/Python/cil/processors/PaganinProcessor.py

```diff
@@ -0,0 +1,405 @@
+
+    return_units: string (optional)
+        The distance units to return the sample thickness in, must be one of 'm', 'cm', 'mm' or 'um' (defults is 'cm') 
+
+    Returns
+    -------
+    AcquisitionData
+        AcquisitionData corrected for phase effects, retrieved sample thickness or (if :code:`full_retrieval=False`) filtered data 
+                
+    Example
+    -------
+    >>> processor = PaganinProcessor(delta=5, beta=0.05, energy=18000)
+    >>> processor.set_input(data)
+    >>> thickness = processor.get_output()
+
+    Example
+    -------
+    >>> processor = PaganinProcessor(delta=1,beta=10e2, full_retrieval=False)
+    >>> processor.set_input(data)
+    >>> filtered_image = processor.get_output()
+
+    Example
+    -------
+    >>> processor = PaganinProcessor()
+    >>> processor.set_input(data)
+    >>> thickness = processor.get_output(override_filter={'alpha':10})
+    >>> phase_retrieved_image = thickness*processor.mu
+
+    References
+    ---------
+    - [1] https://doi.org/10.1046/j.1365-2818.2002.01010.x 
+    - [2] https://iopscience.iop.org/article/10.1088/2040-8986/abbab9
+    - [3] https://henke.lbl.gov/optical_constants/getdb2.html
+    With thanks to Rajmund Mokso for help with the initial implementation of the phase retrieval algorithm
+
+    """
+
+    UNIT_LIST = ['m','cm','mm','um']
+    UNIT_MULTIPLIERS = [1, 1e-2, 1e-3, 1e-6]
+   
+    def __init__(self, delta = 1, beta = 1e-2, energy = 40000,  full_retrieval=True, filter_type='paganin_method', pad = 0, return_units='cm'):
+
+        if return_units not in self.UNIT_LIST:
+            raise ValueError("Return units {} not recognised, expected one of {}, please update data.geometry.config.units".format(str(return_units),str(self.UNIT_LIST)))
+        else:
+            return_multiplier = 1/(self.UNIT_MULTIPLIERS[self.UNIT_LIST.index(return_units)])
+        
+        kwargs = {
+            'energy' : energy,
+            'wavelength' : self._energy_to_wavelength(energy, return_multiplier),
+            'delta': delta,
+            'beta': beta,
+            '_delta_user' : delta,
+            '_beta_user' : beta,
+            'filter_type' : filter_type,
+            'mu' : None,
+            'alpha' : None,
+            'pixel_size' : None,
+            'propagation_distance' : None,
+            'magnification' : None,
+            'filter' : None,
+            'full_retrieval' : full_retrieval,
+            'pad' : pad,
+            'override_geometry' : None,
+            'override_filter' : None,
+            'return_multiplier' : return_multiplier
+            }
+        
+        super(PaganinProcessor, self).__init__(**kwargs)
+
+    def check_input(self, data):
+        if not isinstance(data, (AcquisitionData)):
+            raise TypeError('Processor only supports AcquisitionData')
+      
+        return True
+        
+    def process(self, out=None):
```

Comment:
Code looks good - it will be helpful to add 1-2 lines of comments ahead of each part to explain the flow of what happens and why

Reply:
Thanks Jakob, I've added some comments.